test/jdk/java/awt/print/PrinterJob/InvalidPage.java

```diff
@@ -21,223 +21,121 @@
-         "in the dialog. Repeat for as many printers as you have installed",
-         "On solaris and linux just one printer is sufficient",
-         "Collect the output and examine it, each print job has two pages",
-         "of very similar output, except that the 2nd page of the job may",
-         "appear in a different colour, and the output near the edge of",
-         "the page may be clipped. This is OK. Hold up both pieces of paper",
-         "to the light and confirm that the lines and text (where present)",
-         "are positioned identically on both pages",
-         "The test fails if the JRE crashes, or if the output from the two",
-         "pages of a job is aligned differently"
-       };
-      Sysout.createDialog( );
-      Sysout.printInstructions( instructions );
-
-     new InvalidPage();
-  }
-
-}
-
-class Sysout {
-   private static TestDialog dialog;
-
-   public static void createDialogWithInstructions( String[] instructions )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      dialog.printInstructions( instructions );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-
-   public static void printInstructions( String[] instructions )
-    {
-      dialog.printInstructions( instructions );
+            } catch (PrinterException pe) {
+            }
+        });
+        panel.add(printButton);
+        add(panel);
+        pack();
     }
 
-
-   public static void println( String messageIn )
-    {
-      dialog.displayMessage( messageIn );
+    public int print(Graphics graphics, PageFormat pageFormat, int pageIndex) {
+
+        if (pageIndex > 1) {
+            return Printable.NO_SUCH_PAGE;
+        }
+
+        Graphics2D g2d = (Graphics2D) graphics;
+
+        g2d.translate(pageFormat.getImageableX(), pageFormat.getImageableY());
+        g2d.drawString("ORIGIN", 30, 30);
+        g2d.drawString("X THIS WAY", 200, 50);
+        g2d.drawString("Y THIS WAY", 60, 200);
+        g2d.drawRect(0, 0, (int) pageFormat.getImageableWidth(),
+                (int) pageFormat.getImageableHeight());
+        if (pageIndex == 0) {
+            g2d.setColor(Color.black);
+        } else {
+            g2d.setColor(new Color(0, 0, 0, 128));
+        }
+        g2d.drawRect(1, 1, (int) pageFormat.getImageableWidth() - 2,
+                (int) pageFormat.getImageableHeight() - 2);
+
+        g2d.drawLine(0, 0,
+                (int) pageFormat.getImageableWidth(),
+                (int) pageFormat.getImageableHeight());
+        g2d.drawLine((int) pageFormat.getImageableWidth(), 0,
```

Comment:
dispose g2D object

Reply:
Please, don't dispose of `Graphics` in `print` and `paint` methods where it's passed as a parameter.